mlflow/server/handlers.py

```diff
@@ -16,29 +17,42 @@
     UpdateRun, LogMetric, LogParam, SetTag, ListExperiments, \
     DeleteExperiment, RestoreExperiment, RestoreRun, DeleteRun, UpdateExperiment, LogBatch
 from mlflow.store.artifact_repository_registry import get_artifact_repository
-from mlflow.tracking.utils import _is_database_uri, _is_local_uri
+from mlflow.tracking.utils import TrackingStoreRegistry
 from mlflow.utils.proto_json_utils import message_to_json, parse_dict
 from mlflow.utils.search_utils import SearchFilter
 from mlflow.utils.validation import _validate_batch_log_api_req
 
 _store = None
 
 
+def _get_file_store(store_uri, artifact_uri):
+    from mlflow.store.file_store import FileStore
+    return FileStore(store_uri, artifact_uri)
+
+
+def _get_sqlalchemy_store(store_uri, artifact_uri):
+    from mlflow.store.sqlalchemy_store import SqlAlchemyStore
+    return SqlAlchemyStore(store_uri, artifact_uri)
+
+
+_tracking_store_registry = TrackingStoreRegistry()
+_tracking_store_registry.register('', _get_file_store)
+_tracking_store_registry.register('file', _get_file_store)
+for scheme in DATABASE_ENGINES:
+    _tracking_store_registry.register(scheme, _get_sqlalchemy_store)
+
+
 def _get_store(backend_store_uri=None, default_artifact_root=None):
     from mlflow.server import BACKEND_STORE_URI_ENV_VAR, ARTIFACT_ROOT_ENV_VAR
     global _store
     if _store is None:
-        store_dir = backend_store_uri or os.environ.get(BACKEND_STORE_URI_ENV_VAR, None)
+        store_uri = backend_store_uri or os.environ.get(BACKEND_STORE_URI_ENV_VAR, None)
         artifact_root = default_artifact_root or os.environ.get(ARTIFACT_ROOT_ENV_VAR, None)
-        if _is_database_uri(store_dir):
```

Comment:
I could have done a simpler change of fixing `_is_database_uri` to use `utils.get_uri_scheme`. Using `TrackingStoreRegistry` instead so there is more common code between `MLflowClient` and the CLI server.

Reply:
👍